Remove CSI and rplidar modules (#2279)
viamrobotics/docs: docs/mobility/slam/_index.md

SLAM is an important area of ongoing research in robotics, particularly for mobile applications such as drones, boats, and rovers.

The Viam SLAM service supports the integration of SLAM as a service on your robot.

> You can conduct SLAM with data collected live by a [RPlidar](https://app.viam.com/module/viam/rplidar) or with LIDAR data you provide in configuration, and easily view the map you build on the **SLAM library** tab of your location's page in the [Viam app](https://app.viam.com):

Author: ```suggestion
You can conduct SLAM with data collected live by a [RPlidar](https://github.com/viamrobotics/rplidar) or with LIDAR data you provide in configuration, and easily view the map you build on the **SLAM library** tab of your location's page in the [Viam app](https://app.viam.com):
```

![Completed SLAM maps in the SLAM library tab](/mobility/slam/view-map-page.png)

## Used With

<!-- markdownlint-disable MD034 -->

{{< cards >}}

> {{< relatedcard link="/build/configure/components/camera/" alt_title="RPlidar" required="yes" alt_link="https://github.com/viamrobotics/rplidar">}}

Author: ```suggestion
{{< relatedcard link="/build/configure/components/camera/" alt_title="RPlidar" required="yes">}}
```

SLAM is an important area of ongoing research in robotics, particularly for mobile applications such as drones, boats, and rovers.

The Viam SLAM service supports the integration of SLAM as a service on your robot.
You can conduct SLAM with data collected live by a [RPlidar](https://app.viam.com/module/viam/rplidar) or with LIDAR data you provide in configuration, and easily view the map you build on the **SLAM library** tab of your location's page in the [Viam app](https://app.viam.com):

![Completed SLAM maps in the SLAM library tab](/mobility/slam/view-map-page.png)

## Used With

> 

Author: ```suggestion

<!-- markdownlint-disable MD034 -->

```

SLAM is an important area of ongoing research in robotics, particularly for mobile applications such as drones, boats, and rovers.

The Viam SLAM service supports the integration of SLAM as a service on your robot.
You can conduct SLAM with data collected live by a [RPlidar](https://app.viam.com/module/viam/rplidar) or with LIDAR data you provide in configuration, and easily view the map you build on the **SLAM library** tab of your location's page in the [Viam app](https://app.viam.com):

![Completed SLAM maps in the SLAM library tab](/mobility/slam/view-map-page.png)

## Used With

{{< cards >}}
{{< relatedcard link="ml/vision/vision-odometry/"  required="yes" alt_link="https://app.viam.com/module/viam/rplidar">}}

> {{< relatedcard link="/build/configure/components/camera/" alt_title="RPlidar" required="yes" alt_link="https://app.viam.com/module/viam/rplidar">}}

Author: @npentrel Took your rec for using external url with `alt_link` here but it is giving me a markdownlint error: no bare urls used

Reviewer: Interesting. I don't know a way around it so let's turn off that rule. What's your thinking asking using app module URLs vs The github repo?

Reviewer: I am still wondering why module URLs vs github repo - Andrew asked this too on the PR.

Reviewer: Also `{{< relatedcard link="ml/vision/vision-odometry/"  required="yes" alt_link="https://app.viam.com/module/viam/rplidar">}}` is not showing up. The link is not where that model lives.

Author: @npentrel What do you mean/what can I do to turn off that rule?

Author: Also @npentrel I am going to link that card to GitHub as well as thatmodel page will be removed.

Author: Wait, isn't this part inaccurate? In my understanding it wouldn't be Rplidar or Cartographer / Camera, it would be Rplidar / Cartographer. Want to confirm with you but should I change? @npentrel

Reviewer: I don't know a way to do this without triggering the markdownlint error. You can turn off the markdownlint rule in the .markdownlint settings. But it would be better to turn the rule off for this file only. Please check the markdownlint docs on how to do this. I don't know off the top of my head

Whether the module docs are on the docs site does not affect whether it's required for slam. If it is a requirement you should link to it.

@andf-viam said this about the two options in a previous comment. Probably worth asking the SME to confirm

> {{< relatedcard link="/build/configure/components/camera/" alt_title="RPlidar" required="yes" alt_link="https://app.viam.com/module/viam/rplidar">}}

Author: @npentrel Took your rec for using external url with `alt_link` here but it is giving me a markdownlint error: no bare urls

> {{< relatedcard link="/build/configure/components/camera/" alt_title="RPlidar" required="yes" alt_link="https://app.viam.com/module/viam/rplidar">}}

Author: ```suggestion
{{< relatedcard link="/build/configure/components/camera/" alt_title="RPlidar" required="yes" alt_link=<"https://app.viam.com/module/viam/rplidar">>}}
```

> or

Reviewer: Replace this with:
```suggestion
<div class="align-content-center">or</div>
```

And then add an item to the css file to do

```
.align-content-center {
    align-self: center
}
```

Author: Done-- I think right css file, lmk if not

Reviewer: yep! all good!
